tslearn/matrix_profile.py

```diff
@@ -91,15 +97,27 @@ class MatrixProfile(TransformerMixin,
 
     References
     ----------
-    .. [1] C. M. Yeh, Y. Zhu, L. Ulanova, N.Begum et al. 
-       Matrix Profile I: All Pairs Similarity Joins for Time Series: A 
+    .. [1] C. M. Yeh, Y. Zhu, L. Ulanova, N.Begum et al.
+       Matrix Profile I: All Pairs Similarity Joins for Time Series: A
        Unifying View that Includes Motifs, Discords and Shapelets.
        ICDM 2016.
     """
 
-    def __init__(self, subsequence_length=1, scale=True):
+    def __init__(self, subsequence_length=1, algorithm=None, scale=True):
         self.subsequence_length = subsequence_length
         self.scale = scale
+        self.check_if_algorithm_exists(algorithm)
+        self.algorithm = algorithm
+
+    def check_if_algorithm_exists(self, algorithm):
+        available_algorithms = ["stump"]
+        if (
+            algorithm is not None
+            and algorithm not in available_algorithms
+        ):
+            raise ValueError(
+                "This matrix profile algorithm is not implemented."
```

Comment:
Rather than "not implemented", is it better to say something like `f'This "{algorithm}" matrix profile algorithm is not recognized'`?